### PR TITLE
Add error checking to bcast TSP scatterv_allgatherv algorithm

### PR DESCRIPTION
**Pull request by zhenggb72**
Implement error checking properly for scatterv-allgatherv algorithm

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpi/coll/bcast/bcast_allcomm_nb.c [hunk 1]
```diff
@@ -17,7 +17,10 @@ int MPIR_Bcast_allcomm_nb(void *buffer, MPI_Aint count, MPI_Datatype datatype, i
 
     mpi_errno = MPIC_Wait(req_ptr);
     MPIR_ERR_CHECK(mpi_errno);
+
+    mpi_errno = req_ptr->status.MPI_ERROR;
     MPIR_Request_free(req_ptr);
+    MPIR_ERR_CHECK(mpi_errno);
```
Comment (hzhou) [inline src/mpi/coll/bcast/bcast_allcomm_nb.c:23]: Is this necessary? I believe `MPIC_Wait` will return `MPI_ERR_IN_STATUS` if `req_ptr->status.MPI_ERROR` is not 0.

Comment (zhenggb72) [reply src/mpi/coll/bcast/bcast_allcomm_nb.c:23]: I see. Looks like we didn't get your commit 46921feda06ac29db9d2c6ffe71e6646ea9d13ef, where mpi_errno is set from req_ptr->status.

Comment (zhenggb72) [reply src/mpi/coll/bcast/bcast_allcomm_nb.c]: removed this commit.

**Review by hzhou — APPROVED** [review]
LGTM. Please do the author checklist and clear the tests.

**Comment by hzhou** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

**Comment by zhenggb72** [comment]
just rebased the branch. Can merge now?

**Comment by hzhou** [comment]
> just rebased the branch. Can merge now?

Did the tests clear? We saw a ucx regression in a separate PR, just like to confirm if it's sporadic.

**Comment by zhenggb72** [comment]
Yes, all tests were clear.

**Pull request merged**